File: julia/etl/etl.jl
function transform(input::AbstractDict)
    # take 1, with explicit loops
    # 
    # result = Dict{AbstractChar, Int}()
    # for (value, letters) in pairs(input)
    #     for letter in letters
    #         result[lowercase(letter)] = value
    #     end
    # end
    # return result

    # take 2, with comprehensions
    Dict(
        lowercase(letter) => value
        for (value, letters) in pairs(input) 
        for letter in letters
    )
end

File: go/etl/etl.go
package etl

import "strings"

func Transform(in map[int][]string) map[string]int {
	out := map[string]int{}

	for val, letters := range in {
		for _, letter := range letters {
			out[strings.ToLower(letter)] = val
		}
	}

	return out
}
